lib/httpoison/base.ex

```diff
@@ -394,24 +394,26 @@ defmodule HTTPoison.Base do
 
     case :hackney.request(method, request_url, request_headers,
                           request_body, hn_options) do
-      {:ok, status_code, headers, _client} when status_code in [204, 304] ->
-        response(process_status_code, process_headers, process_response_body, status_code, headers, "")
-      {:ok, status_code, headers} -> response(process_status_code, process_headers, process_response_body, status_code, headers, "")
+      {:ok, status_code, headers, client} when status_code in [204, 304] ->
+        response(process_status_code, process_headers, process_response_body, status_code, headers, "", :hackney.location(client))
+      {:ok, status_code, headers} -> response(process_status_code, process_headers, process_response_body, status_code, headers, "", nil)
       {:ok, status_code, headers, client} ->
+        location = :hackney.location(client) # FIXME The current version of Hackney (0.8) requires that we read the location before the body.
```

Comment:
@benoitc, do you mean this?

Reply:
yeah probably :) i need to think on a better way to keep that state..

Reply:
So does that mean I should not rework this PR to make it work against the current httpoison codebase and wait?

Reply:
@edgurgel @benoitc so what do I do here? Should I rebase or wait?

Reply:
I think we just keep moving as this has been open for a while. We can always rework it if hackney changes in the future. Thanks for pushing this :D